File: leetcode/js/left-and-right-sum-differences.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRightDifference = function (nums) {
  let leftSum = 0;
  let rightSum = nums.reduce((acc, cur) => acc + cur, 0);

  return nums.map((num) => {
    rightSum -= num;
    const result = Math.abs(leftSum - rightSum);
    leftSum += num;
    return result;
  });
};

File: leetcode/rust/left-and-right-sum-differences.rs
impl Solution {
    pub fn left_right_difference(nums: Vec<i32>) -> Vec<i32> {
        let mut left_sum: i32 = 0;
        let mut right_sum = nums.iter().sum();

        nums.iter()
            .map(|&num| {
                right_sum -= num;
                let res = left_sum.abs_diff(right_sum);
                left_sum += num;
                res as i32
            })
            .collect::<Vec<_>>()
    }
}
